test/sources/test_registry.ts

```
@@ -0,0 +1,478 @@
+
+  it("ignores array values in fetcherParams (schema mismatch)", () => {
+    const f = base();
+    f.set("weird_array", ["a", "b"]);
+    const source = buildSource(f, "");
+    assert.ok(source);
+    assert.equal(source!.fetcherParams["weird_array"], undefined);
+  });
+});
+
+describe("serializeSource — round trip", () => {
+  function makeSource(over: Partial<Source> = {}): Source {
```

ESLint

[error] 'makeSource' is already declared in the upper scope on line 319 column 10.

(no-shadow)